fastapi/applications.py

```diff
@@ -769,6 +769,7 @@ def trace(
     def run(self, **args):
         try:
             import uvicorn
-            uvicorn.run(self, **args)
         except ImportError:
```

Comment:
```suggestion
        except ModuleNotFoundError:
```

Reply:
Using this suggestion, this will raise a more informative diagnostic if uvicorn is installed, but fails to import

Reply:
Traceback (most recent call last):
  File "C:\Users\Isaac\stdinmedia\uvicorn.py", line 11, in <module>
    app.run()
  File "C:\Users\Isaac\stdinmedia\venv\lib\site-packages\fastapi\applications.py", line 771, in run
    import uvicorn
  File "C:\Users\Isaac\stdinmedia\uvicorn.py", line 11, in <module>
    app.run()
  File "C:\Users\Isaac\stdinmedia\venv\lib\site-packages\fastapi\applications.py", line 775, in run
    uvicorn.run(self, **args)
AttributeError: partially initialized module 'uvicorn' has no attribute 'run' (most likely due to a circular import)


This is the error that is raised when I name the project file uvicorn. It is neither an ImportError or a ModuleNotFoundError. I really don't see it.

Reply:
Here's some more details behind the motivation of `ModuleNotFoundError`: https://bugs.python.org/issue15767

if `import uvicorn` raises an instance of `ImportError` that's not an instance of `ModuleNotFoundError` then installing uvicorn won't fix it

Reply:
I have just tested it, and see no difference. I'll just switch it I guess.